crates/prover/src/da_block_handler.rs

```diff
@@ -347,10 +348,10 @@ where
                 sequencer_commitment_state_diff.clone(),
             );
 
-            let serialized_state_diff = borsh::to_vec(&cumulative_state_diff)?;
+            let compressed_state_diff = compress_blob(&borsh::to_vec(&cumulative_state_diff)?);
```

Comment:
Maybe also explain why we want to check the compressed version in a comment rather than uncompressed one,